### DISCUSSION TITLE
Incompatibility between Application & bodyparser

### DISCUSSION THREAD

**Discussion by DraftProducts** [adonisjs/core#2785] [Help] [answered]
Hi everyone !
I upgraded from preview to latest versions and it looks like that the bodyparser middlewere is registred with a old/inexistant method.

```js
[1623578325185] ERROR (backend/42 on 6143516ac35e): application_1.inject is not a function
```

```js
// bodyparser/build/src/BodyParser/index.js L226-230
BodyParserMiddleware = __decorate([
    application_1.inject(['Adonis/Core/Config']),
    __metadata("design:paramtypes", [Object])
], BodyParserMiddleware);
exports.BodyParserMiddleware = BodyParserMiddleware;
```

In Ts bodyparser the inject method is invoked by a decorator.

https://github.com/adonisjs/bodyparser/blob/master/src/BodyParser/index.ts#L30
![image](https://user-images.githubusercontent.com/29425008/121803250-ef517b80-cc40-11eb-8269-48d2abbc8527.png)

I see that the peer dep version of Application is not the same than the dev dep :thinking: 
https://github.com/adonisjs/bodyparser/blob/master/package.json#L56
```json
"devDependencies": {
  "@adonisjs/application": "^5.1.3",
},
"peerDependencies": {
  "@adonisjs/application": "^5.0.0",
},
```

Does some one have a idea ? Is a issue needed ?

**Comment by DraftProducts** [comment, marked as answer]
I looked everywhere but not in my own project because i was sure that i updated all deps with the 
https://docs.adonisjs.com/releases/out-of-preview#upgrading-to-the-latest-versions article.

The @adonisjs/application wasn't upgraded.

Hi @thetutlage, i think that it will be a good idea to add `npm i @adonisjs/application@latest` to the deps list.

**Reply by thetutlage** [reply]
The `@adonisjs/application` is the dependency of `@adonisjs/core` and you should not install it separately inside your app
